app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -509,6 +521,56 @@ public void onTabReselected(TabLayout.Tab tab) {
         });
 
         tabs.selectTab(tabs.getTabAt(0));
+
+        mCropperLauncher = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(),
+                new ActivityResultCallback<ActivityResult>() {
+                    @Override
+                    public void onActivityResult(ActivityResult result) {
+                        Intent intent = result.getData();
+                        if (result.getResultCode() == Activity.RESULT_OK) {
+                            Uri debugUri = UCrop.getOutput(intent);
+                            String cropOutputPath = debugUri.getPath();
+                            Log.d("cropper", "cropper thinks it has produced image at " + cropOutputPath + " " + debugUri.toString());
+                            Bitmap bitmap = BitmapFactory.decodeFile(cropOutputPath);
+
+
+                            if (bitmap != null) {
+                                bitmap = Utils.resizeBitmap(bitmap);
+                                try {
+                                    bitmap = Utils.rotateBitmap(bitmap, new ExifInterface(cropOutputPath));
+                                } catch (IOException e) {
+                                    e.printStackTrace();
+                                }
+
+                                if (mRequestedImage == Utils.CARD_IMAGE_FROM_CAMERA_FRONT || mRequestedImage == Utils.CARD_IMAGE_FROM_FILE_FRONT) {
+                                    setCardImage(cardImageFront, bitmap);
+                                } else {
+                                    setCardImage(cardImageBack, bitmap);
+                                }
+
+                                hasChanged = true;
+                            } else {
+                                Toast.makeText(LoyaltyCardEditActivity.this, R.string.errorReadingImage, Toast.LENGTH_LONG).show();
+                            }
+                        }else if(result.getResultCode() == UCrop.RESULT_ERROR){
+                            Log.e("cropper error", UCrop.getError(intent).toString());
+                        }
+                    }
+                });
+
+        if (savedInstanceState != null) {
+            skipOnce = savedInstanceState.getInt("skipOnce") == 1 ? true : false;
+        }
+        mCropperOptions = new UCrop.Options();
+        mCropperOptions.setCompressionFormat(Bitmap.CompressFormat.JPEG);
```

Comment:
```suggestion
        mCropperOptions.setCompressionFormat(Bitmap.CompressFormat.JPEG);
        mCropperOptions.setFreeStyleCropEnabled(true);
```
Enables freely changing the Aspect Ratio